### PR TITLE
[sn-platform][sn-platform-slim]feat: add logConfig support on sn-platform & sn-platform slim charts

### PR DESCRIPTION
**Pull request by ericsyh**
*(If this PR fixes a github issue, please add `Fixes #<xyz>`.)*

Fixes #<xyz>

*(or if this PR is one task of a github issue, please add `Master Issue: #<xyz>` to link to the master issue.)*

Master Issue: #<xyz>

### Motivation

Add the `logConfig` CR configuration support on chart to support dynamic log config changing. 

### Modifications

*Describe the modifications you've done.*

### Verifying this change

- [x] Make sure that the change passes the CI checks.

*(Please pick either of the following options)*

This change is a trivial rework / code cleanup without any test coverage.

*(or)*

This change is already covered by existing tests, such as *(please describe tests)*.

*(or)*

This change added tests and can be verified as follows:

*(example:)*
  - *Added integration tests for end-to-end deployment with large payloads (10MB)*
  - *Extended integration test for recovery after broker failure*

### Documentation

Check the box below.

Need to update docs? 

- [x] `doc-required` 
  
  (If you need help on updating docs, create a doc issue)
  
- [ ] `no-need-doc` 
  
  (Please explain why)
  
- [ ] `doc` 
  
  (If this PR contains doc changes)

### CODE REVIEW

File: examples/sn-platform/logConfig.yaml [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+  # The field logConfig can be used to change the log level and log format of pods.
+  # The logConfig field is optional. If it is not specified, the component will use the default log configuration /pulsar/conf/log4j2.yaml. 
+  # If it is specified will dynamically change the log level and log format of the component by changing the CR.
+  logConfig:
+  # The level field can be used to change the log level of the component. The value can be one of TRACE, DEBUG, INFO, WARN, ERROR, FATAL, OFF.
+    level: "DEBUG"
+  # The format field can be used to change the log format of the component. The value can be one of json, text.
+    format: "json"
+  # The template field can totally change the log config of the component. The value is a string, which is the content of the log config file.
+    template:
```
Comment (labuladong) [inline examples/sn-platform/logConfig.yaml:30]:
```suggestion
    template: |
```

File: examples/pulsar-operators/logConfig.yaml [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+  # The field logConfig can be used to change the log level and log format of pods.
+  # The logConfig field is optional. If it is not specified, the component will use the default log configuration /pulsar/conf/log4j2.yaml. 
+  # If it is specified will dynamically change the log level and log format of the component by changing the CR.
+  logConfig:
+  # The level field can be used to change the log level of the component. The value can be one of TRACE, DEBUG, INFO, WARN, ERROR, FATAL, OFF.
+    level: DEBUG
+   # The format field can be used to change the log format of the component. The value can be one of json, text.
+    format: json
+  # The template field can totally change the log config of the component. The value is a string, which is the content of the log config file.
+    template: |
```
Comment (labuladong) [inline examples/pulsar-operators/logConfig.yaml:35]:
```suggestion
    template:
```

File: charts/sn-platform/values.yaml [hunk 3]
```diff
@@ -1065,6 +1087,17 @@ broker:
     # annotations that will be added on the PulsarBroker CR only.
     annotations: {}
 
+  # The field logConfig can be used to change the log level and log format of pods.
+  # The logConfig field is optional. If it is not specified, the component will use the default log configuration /pulsar/conf/log4j2.yaml.
```
Comment (labuladong) [inline charts/sn-platform/values.yaml:1091]:
```suggestion
  # The logConfig field is optional. If it is not specified, the component will use the default log configuration /pulsar/conf/log4j2.yaml.
```

File: charts/sn-platform/values.yaml [hunk 4]
```diff
@@ -1444,6 +1477,17 @@ proxy:
     # annotations that will be added on the PulsarProxy CR only.
     annotations: {}
 
+  # The field logConfig can be used to change the log level and log format of pods.
+  # The logConfig field is optional. If it is not specified, the component will use the default log configuration /pulsar/conf/log4j2.yaml.
```
Comment (labuladong) [inline charts/sn-platform/values.yaml:1481]:
```suggestion
  # The logConfig field is optional. If it is not specified, the component will use the default log configuration /pulsar/conf/log4j2.yaml.
```

**Comment by ericsyh** [comment]
@labuladong PTAL again.

**Review by labuladong — APPROVED** [review]
LGTM

**Pull request merged**